packages/tpetra/core/test/CrsMatrix/Bug8447.cpp

```diff
@@ -0,0 +1,148 @@
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Michael A. Heroux (maherou@sandia.gov)
+//
+// ************************************************************************
+// @HEADER
+*/
+
+// This bug is described at
+// https://github.com/trilinos/Trilinos/issues/8447
+//
+#include <Tpetra_ConfigDefs.hpp>
+#include <Teuchos_UnitTestHarness.hpp>
+#include <TpetraCore_ETIHelperMacros.h>
+#include <Tpetra_Map.hpp>
+#include <Tpetra_CrsMatrix.hpp>
+#include <Tpetra_TestingUtilities.hpp>
+
+#include <iostream>
+#include <vector>
+
+namespace { // anonymous
+
+using Teuchos::ArrayView;
+using Teuchos::RCP;
+using Teuchos::rcp;
+using Teuchos::Comm;
+using Teuchos::OrdinalTraits;
+using Tpetra::TestingUtilities::getDefaultComm;
+
+TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL(CrsMatrix, Bug8447, SC, LO, GO, NT)
+{
+
+  typedef Tpetra::global_size_t GST; // Map's constructor needs this
+
+  typedef Tpetra::Map<LO,GO,NT> MapType;
+  typedef Tpetra::Import<LO,GO,NT> ImportType;
+  typedef Tpetra::CrsMatrix<SC,LO,GO,NT> CrsMatrixType;
+
+  RCP<const Comm<int> > comm = getDefaultComm();
+  TEUCHOS_TEST_FOR_EXCEPTION(
+    comm->getSize () != 2, std::logic_error,
+    "This test must be run with exactly 2 MPI processes.");
+
+  const int myPID = comm->getRank();
+
+  std::vector<GO> globalIDs, globalIDs2;
+  std::vector< std::vector<GO> > indices;
+  std::vector< std::vector<SC> > values;
+  if (myPID == 0) {
+    globalIDs.push_back(0);
+    globalIDs.push_back(1);
+    globalIDs.push_back(2);
+    globalIDs2.push_back(0);
+    globalIDs2.push_back(1);
+    globalIDs2.push_back(2);
+    globalIDs2.push_back(3);
+    globalIDs2.push_back(4);
+
+    values.resize(3);
+    values[0].resize(1, 1);
+    values[1].resize(1, 1);
+    values[2].resize(1, 1);
+    indices.resize(3);
+    indices[0].push_back(0); indices[0].push_back(1);
+    indices[1].push_back(1); indices[1].push_back(2);
+    indices[2].push_back(2);
+  }
+  else {
+    globalIDs.push_back(3);
+    globalIDs.push_back(4);
+    values.resize(2);
+    values[0].resize(1, 1);
+    values[1].resize(1, 1);
+    indices.resize(2);
+    indices[0].push_back(3); indices[0].push_back(4);
+    indices[1].push_back(4);
+  }
+
+  const GST INVALID = OrdinalTraits<GST>::invalid ();
+  const GO indexBase = 0;
+  RCP<const MapType> rowMap =
+    rcp (new MapType (INVALID, ArrayView<GO> (globalIDs), indexBase, comm));
+  RCP<const MapType> rowMap2 =
+    rcp (new MapType (INVALID, ArrayView<GO> (globalIDs2), indexBase, comm));
+  TEUCHOS_TEST_FOR_EXCEPTION(
+    ! rowMap->isOneToOne (), std::logic_error,
+    "In this test, the row Map is supposed to be one to one.");
+
+  RCP<CrsMatrixType> matrix = rcp (new CrsMatrixType (rowMap, 2));
+  for (size_t i = 0; i < static_cast<size_t> (globalIDs.size ()); ++i) {
+    matrix->insertGlobalValues (globalIDs[i],
```

Comment:
@cgcgcg This adds to the UVM backlog.  @trilinos/tpetra @csiefer2 What's the canonical way to do this without UVM?

Reply:
Ugh, indeed. Lemme rewrite this with views.